src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,201 @@
+                    "org.apache.spark.api.python.PythonRDD",
+                    "setupBroadcast",
+                    path);
+
+                if (sc.EncryptionEnabled)
+                {
+                    throw new NotImplementedException(
+                            "Broadcast encryption is not supported yet.");
+                }
+                else
+                {
+                    WriteBroadcastValueToFile(sc, path, value);
+                }
+
+                _jvmObject = (JvmObjectReference)sparkContext.Jvm.CallStaticJavaMethod(
+                    "org.apache.spark.sql.api.dotnet.SQLUtils",
+                    "createBroadcast",
+                    sparkContext,
+                    pythonBroadcast);
+            }
+            _bid = (long)_jvmObject.Invoke("id");
+            JvmBroadcastRegistry.s_jvmBroadcastVariables.Add(_jvmObject);
+        }
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Get the broadcasted value.
+        /// </summary>
+        /// <returns>The broadcasted value</returns>
+        public T Value()
+        {
+            return (T)BroadcastRegistry.s_registry[_bid];
+        }
+
+        /// <summary>
+        /// Asynchronously delete cached copies of this broadcast on the executors.
+        /// If the broadcast is used after this is called, it will need to be re-sent to each
+        /// executor.
+        /// </summary>
+        public void Unpersist()
+        {
+            _jvmObject.Invoke("unpersist");
+        }
+
+        /// <summary>
+        /// Delete cached copies of this broadcast on the executors. If the broadcast is used after
+        /// this is called, it will need to be re-sent to each executor.
+        /// </summary>
+        /// <param name="blocking">Whether to block until unpersisting has completed</param>
+        public void Unpersist(bool blocking)
+        {
+            _jvmObject.Invoke("unpersist", blocking);
+        }
+
+        /// <summary>
+        /// Destroy all data and metadata related to this broadcast variable. Use this with
+        /// caution; once a broadcast variable has been destroyed, it cannot be used again.
+        /// This method blocks until destroy has completed.
+        /// </summary>
+        public void Destroy()
+        {
+            _jvmObject.Invoke("destroy");
+        }
+
+        /// <summary>
+        /// Function that creates a file to store the broadcast value in the given path.
+        /// </summary>
+        /// <param name="sc">Spark Context object</param>
+        /// <param name="path">Path where file is to be created</param>
+        /// <param name="value">Broadcast value to be written to the file</param>
+        private void WriteBroadcastValueToFile(SparkContext sc, string path, object value)
+        {
+            Directory.CreateDirectory(sc.TempDir);
```

Comment:
I don't understand why you need to get the temp dir from sc.

Reply:
I will set it to derive it at the constructor of Broadcast. Thanks for the suggestion!